src/test/test_verify_multithreading.cpp

```
@@ -13,31 +24,78 @@ static void test_analyze_thread(const prevail::Program* prog, bool* res) {
+    REQUIRE(res2);
+}
+
+// Two programs from the same ELF must share the immutable platform reference
+// but have fully disjoint per-program environments and analysis-prep facts.
+// If the PlatformSpec / ProgramEnvironment split is ever reintroduced as shared
+// mutable state this test will fail.
+TEST_CASE("multi-program sharing invariant", "[verify][multithreading]") {
+    const prevail::Program prog1 = prepare("2/1");
+    const prevail::Program prog2 = prepare("2/2");
+
+    // Immutable platform is shared across programs.
```

Cppcheck (2.20.0)

[performance] Function parameter 'variable_terms' should be passed by const reference.

(passedByValue)